test/honey/sql/helpers_test.cljc

```diff
@@ -866,3 +866,27 @@
            {:with [[:a]],
             :insert-into [[:quux [:x :y]]
                           {:select [:id], :from [:table]}]}))))
+
+(deftest quoting-:%-syntax
```

Comment:
Since this is really about testing the core `%` syntax, it should be in `sql_test.cljc` (and not use the helpers to construct the expression -- just `{:select [...]}` is sufficient.